core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -335,8 +335,8 @@ object Pull extends PullLowPriority {
         f: Chunk[O] => Pull[F2, O2, Unit]
     ): Pull[F2, O2, Unit] =
       uncons.flatMap {
-        case None           => Pull.done
-        case Some((hd, tl)) => f(hd) >> tl.unconsFlatMap(f)
+        case Some(x) => f(x._1) >> x._2.unconsFlatMap(f)
+        case _       => Pull.done
       }
```

Comment:
Before:
```java
public final <F2, O2, F, O> Pull<F2, O2, BoxedUnit> unconsFlatMap$extension(Pull<F, O, BoxedUnit> $this, Function1<Chunk<O>, Pull<F2, O2, BoxedUnit>> f) {
    return this.uncons$extension($this).flatMap((Function1 & Serializable)x0$1 -> {
        Some some;
        Tuple2 tuple2;
        Option option = x0$1;
        if (None$.MODULE$.equals(option)) {
            return Pull$.MODULE$.done();
        }
        if (option instanceof Some && (tuple2 = (Tuple2)(some = (Some)option).value()) != null) {
            Chunk hd = (Chunk)tuple2._1();
            Pull tl = (Pull)tuple2._2();
            return ((Pull)f.apply((Object)hd)).$greater$greater((Function0 & Serializable)() -> MODULE$.unconsFlatMap$extension(Pull$.MODULE$.StreamPullOps(tl), f));
        }
        throw new MatchError((Object)option);
    });
}
```

After:
```java
public final <F2, O2, F, O> Pull<F2, O2, BoxedUnit> unconsFlatMap$extension(Pull<F, O, BoxedUnit> $this, Function1<Chunk<O>, Pull<F2, O2, BoxedUnit>> f) {
    return this.uncons$extension($this).flatMap((Function1 & Serializable)x0$1 -> {
        Option option = x0$1;
        if (option instanceof Some) {
            Some some = (Some)option;
            Tuple2 x = (Tuple2)some.value();
            return ((Pull)f.apply(x._1())).$greater$greater((Function0 & Serializable)() -> MODULE$.unconsFlatMap$extension(Pull$.MODULE$.StreamPullOps((Pull)x._2()), f));
        }
        return Pull$.MODULE$.done();
    });
}
```

Reply:
(Almost) same code as well, why `(tuple2 = (Tuple2)(some = (Some)option).value()) != null)`?
Also, would it make sense to keep `Some` higher in case match? (happy path?)